src/testrender/optixraytracer.cpp

```diff
@@ -1398,21 +1326,130 @@ OptixRaytracer::render(int xres OSL_MAYBE_UNUSED, int yres OSL_MAYBE_UNUSED)
     params.invh = 1.0f / m_yres;
     params.output_buffer = d_output_buffer;
     params.traversal_handle = m_travHandle;
+    params.osl_printf_buffer_start = d_osl_printf_buffer;
+    // maybe send buffer size to CUDA instead of the buffer 'end'
+    params.osl_printf_buffer_end = d_osl_printf_buffer + OSL_PRINTF_BUFFER_SIZE;
+    params.color_system          = d_color_system;
+    params.test_str_1            = d_test_str_1;
+    params.test_str_2            = d_test_str_2;
 
     CUDA_CHECK (cudaMemcpy (reinterpret_cast<void *>(d_launch_params), &params, sizeof(RenderParams), cudaMemcpyHostToDevice));
 
+    // Set up global variables
+    OPTIX_CHECK (optixLaunch (m_optix_pipeline,
```

Comment:
Is there some reason this can't be done in the regular raygen program? I tried it and all of the tests pass, except for the expected string ops failure.

Reply:
In this particular case, I don't think we need the separate launch to set up globals.  I just figured I'd keep the same logic as from `testshade` in case someone wanted to use `testrender` as an example program on writing an optix-based ray tracer with OSL support.